### DISCUSSION TITLE
icub fingers position/velocity control

### DISCUSSION THREAD

**Discussion by gabrielegiudici93** [robotology/community#504] [Q&A] [answered]
Hi, I'm using the iCub's arm and I noticed that the dynamics of the fingers are dramatically slower wrt the rest of the arm. 
Using yarpmotorgui for example I can change the velocity of the position controller, instead, although I use the maximum velocity ref.(100) on the fingers,I cannot get a faster movement. 
Let me explain myself better: my problem consists in quickly release an object to perform a throwing task, but the time that the robot spend to open the hand results longer than the upper arm movement. 

How could I overcome this problem? Is it possible to increase the reference over 100? Also, I'm looking for a manual of the hand, but i can't find it on the wiki.icub (http://wiki.icub.org/wiki/Manual), is it avaiable?

**Comment by pattacini** [comment, marked as answer]
Hi @gabrielegiudici93 and welcome to Robotology 👋🏻

For how to control the fingers, I would recommend that you peruse this [tutorial][tutorial], which will guide you through the three main modalities you can employ to control joints kinematically:
- `position`
- `velocity`
- `positiondirect`

The tutorial showcases how to control the elbow, but the same applies to the fingers.

To address your task, my suggestion is to write up a simple YARPed module (or adapt the tutorial) so that you can fiddle with the quantities at stake (e.g. maximum acceleration, speed...) plus you will be able to acquire and dump how the joints itself will respond to your commands. The latter turns out to be essential for understanding what's going on in your setup.

That said, fingers are beasts per se, meaning that they do suffer from slack and often require tensioning of the cables. Hence, the general status of the hand of your robot certainly plays a role here.

Finally, by relying on the logs and knowing that the status of the fingers is ok-ish, you can think of tuning the low-level PID gains with the goal of increasing the performance.

Hope this helps.

[tutorial]: https://github.com/vvv-school/tutorial_joint-interface

**Comment by gabrielegiudici93** [comment]
Hi @pattacini, I'm also controlling the arm in force. Thanks for the advice, probably the problem is correlated with the slack of tendons. Could you share with me the instruction/manual to calibrate them? 

Thanks in advance

**Reply by pattacini** [reply]
This is the [guide][1]. I believe yours is a V1 robot.

[1]: https://icub-tech-iit.github.io/documentation/icub_tendons/lower_arm/
